app/Livewire/Project/Service/FileStorage.php

```
@@ -86,6 +90,60 @@ public function syncData(bool $toModel = false): void
+
+                    if ($target === $mountPath && $source && preg_match('/\$\{([a-zA-Z_][a-zA-Z0-9_]*)/', $source, $matches)) {
+                        return $matches[1];
+                    }
+                }
+            }
+        } catch (\Throwable $e) {
+            // Silently fail — this is just a UI hint
+        }
+
+        return null;
+    }
```

PHPMD (2.15.0)

[warning] The method detectEnvVarSource() has a Cyclomatic Complexity of 12. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)